Java/libraries/recognizers-text-number-with-unit/src/main/java/com/microsoft/recognizers/text/numberwithunit/parsers/BaseCurrencyParser.java

```diff
@@ -77,13 +79,16 @@ private ParseResult mergeCompoundUnit(ExtractResult compoundResult) {
                 result = new ParseResult(extractResult.getStart(), extractResult.getLength(), extractResult.getText(), extractResult.getType(), null, null, null);
 
                 mainUnitValue = unitValue;
-                numberValue = parseResultValue.isPresent() ? Double.parseDouble(parseResultValue.get().number) : 0;
+                if (parseResultValue.isPresent() && parseResultValue.get().number != null) {
```

Comment:
What's the benefit of this instead of the previous line?

Reply:
> What's the benefit of this instead of the previous line?

It could be null and break the parser.